src/types/quota.rs

```diff
@@ -0,0 +1,246 @@
+use crate::error::{Error, ParseError};
+use crate::parse::{parse_many_into2, parse_until_done, MapOrNot, MapOrNot2};
+use crate::types::UnsolicitedResponse;
+use imap_proto::Response;
+use ouroboros::self_referencing;
+use std::borrow::Cow;
+use std::fmt::{Debug, Display, Formatter};
+use std::sync::mpsc;
+
+/// From [SETQUOTA Resource limit](https://datatracker.ietf.org/doc/html/rfc2087#section-4.1)
+///
+/// Used by [`Session::set_quota`]
+#[derive(Debug, Eq, PartialEq)]
```

Comment:
Is there a good reason not to derive `Clone` here?

Reply:
I don't think so.  I'll derive it.  the QuotaResourceName is Cow internally when needed so clone will be "Cheap" unless a full clone is needed.